tests/backup/backup_test.go

```diff
@@ -3494,35 +3451,21 @@ var _ = Describe("{DeleteSharedBackup}", func() {
 			err := Inst().S.Destroy(contexts[i], opts)
 			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
 		}
-		err := backup.DeleteUser(userName)
-		log.FailOnError(err, "Error deleting user %v", userName)
-
-		ctx, err := backup.GetAdminCtxFromSecret()
-		log.FailOnError(err, "Fetching px-central-admin ctx")
-
-		log.Infof("Deleting registered clusters for admin context")
-		err = DeleteCluster(SourceClusterName, orgID, ctx)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
-		err = DeleteCluster(destinationClusterName, orgID, ctx)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", destinationClusterName))
 
 		log.Infof("Deleting registered clusters for non-admin context")
 		for _, ctxNonAdmin := range userContexts {
-			err = DeleteCluster(SourceClusterName, orgID, ctxNonAdmin)
+			err := DeleteCluster(SourceClusterName, orgID, ctxNonAdmin)
 			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", SourceClusterName))
 			err = DeleteCluster(destinationClusterName, orgID, ctxNonAdmin)
 			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cluster %s", destinationClusterName))
 		}
 
-		log.Infof("Cleaning up backup location - %s", backupLocationName)
-		err = DeleteBackupLocation(backupLocationName, backupLocationUID, orgID)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", backupLocationName))
+		err := backup.DeleteUser(userName)
+		log.FailOnError(err, "Error deleting user %v", userName)
 
-		log.Infof("Cleaning cloud credential")
-		//TODO: Eliminate time.Sleep
-		time.Sleep(time.Minute * 3)
-		err = DeleteCloudCredential(credName, orgID, cloudCredUID)
-		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting cloud cred %s", credName))
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		DeleteCloudAccounts(backupLocationMap, credName, cloudCredUID, ctx)
```

Comment:
It looks like log.Infof("Cleaning cloud accounts") before DeleteCloudAccounts is missing.